crates/cxx-qt-lib-headers/include/qset.h

```diff
@@ -0,0 +1,100 @@
+// clang-format off
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// clang-format on
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+#pragma once
+
+#include <QtCore/QSet>
+
+#include <QtCore/QDate>
+#include <QtCore/QDateTime>
+// #include <QtCore/QPoint>
```

Comment:
Only `qHash`able types are supported by QSet due to it's internal implementation.

As seen in https://doc.qt.io/qt-6/qhash.html#related-non-members QPoint, QPointF, QRect, QRectF, QSize, QSizeF, QVariant, and QColor are missing.

So should we not implement these ? or should we create a qHash function for them ?

Reply:
Removed these for now

Reply:
Might even need to be behind a feature as we have a few differing issues

- If we implement `qHash` for the types, does this conflict with any `qHash` implementations the developer has made?
- If we require the developer to implement `qHash` then they can't `impl QSetElement for QPoint` as both are outside their crate (they would have to create `MyPoint` and do it that way)
- So maybe we need to have a `assume-qhashed-types` feature which assumes the developer has made a `qHash` method (and therefore all Qt types in cxx-qt-lib are hashable) and is opt-in